### ISSUE TITLE
Extension fails to load in GNOME 3.32

### ISSUE THREAD

**Issue by ronjouch** [pixel-saver/pixel-saver#195]
The usual! 😄, GNOME isn't exactly friendly with extensions backwards compat, eh 😕.

Here are the errors I see in `journalctl`:

```
gnome-shell[19665]: JS ERROR: TypeError: label is undefined
                    onAppMenuHover/tooltipDelayCallbackID<@/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/app_menu.js:116:8
gnome-shell[19665]: [pixel-saver]: 'appmenu' is not a valid button.
gnome-shell[19665]: JS WARNING: [/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/decoration.js 216]: assignment to undeclared variable success
gnome-shell[19665]: JS WARNING: [/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/decoration.js 216]: assignment to undeclared variable pid
gnome-shell[19665]: Some code called array.toString() on a Uint8Array instance. Previously this would have interpreted the bytes of the array as a string, but that is nonstandard. In the future this will return the bytes as comma-separated digits. For the time being, the old behavior has been preserved, but please fix your code anyway to explicitly call ByteArray.toString(array).
                    (Note that array.toString() may have been called implicitly.)
                    0 getOriginalState() ["/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/decoration.js":154]
                    1 setHideTitlebar() ["/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/decoration.js":203]
                    2 onWindowAdded/<() ["/home/ronj/.local/share/gnome-shell/extensions/pixel-saver@deadalnix.me/decoration.js":297]
```

I'm using the latest version available on extensions.gnome.org.

**EDIT**: new extension https://extensions.gnome.org/extension/1732/gtk-title-bar/ ([github](https://github.com/velitasali/GTKTitleBar)) seems to do the job.

**Comment by Norbert01** [comment]
From a no-title-bar (similar extension) request:
[](
https://github.com/franglais125/no-title-bar/issues/113)

> In GNOME 3.32 _GTK_HIDE_TITLEBAR_WHEN_MAXIMIZED is gone: https://gitlab.gnome.org/GNOME/mutter/merge_requests/221
> 
> They say that extensions should use _MOTIF_WM_HINTS instead.

**Comment by jonian** [comment]
I have fixed this issue in my extension (unite) and this is the final solution:
 https://github.com/hardpixel/unite-shell/blob/master/unite%40hardpixel.eu/modules/windowDecoration.js#L86

More info on possible problems using `_MOTIF_WM_HINTS`:
https://gitlab.gnome.org/GNOME/mutter/merge_requests/221#note_461106

**Comment by feimosi** [comment]
For me it fails with the following error:

```
TypeError: global.screen is undefined
```

**Comment by mahdiG** [comment]
@feimosi https://github.com/pixel-saver/pixel-saver/issues/178#issuecomment-423320666

**Comment by ronjouch** [comment]
In addition to already-mentioned extension "Unite" but may do too invasive (as it tries to replicate Ubuntu's Unity feel), I just found https://github.com/velitasali/GTKTitleBar / https://extensions.gnome.org/extension/1732/gtk-title-bar/ , which seems to do the job.

**Comment by matthijskooijman** [comment]
Note that the `global.screen` error first appears in 3.30, is the subject of #195 but is not released yet (#177).

**Comment by tallero** [comment]
> For me it fails with the following error:
> 
> ```
> TypeError: global.screen is undefined
> ```

Install the application not from extensions website (see #177) but from this repository.

> In addition to already-mentioned extension "Unite" (which works but may be too invasive, as it tries to replicate Ubuntu's Unity feel), I just found extension [GTK Title Bar - Remove Title Bar for non-GTK Software](https://github.com/velitasali/GTKTitleBar) / https://extensions.gnome.org/extension/1732/gtk-title-bar/ , which is weirdly named but seems to do the job very well under 3.32 .

GNOME 3.32 has been merged in #199.
